relayer/app/cursor/cursors.go

```diff
@@ -0,0 +1,156 @@
+package cursor
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/expbackoff"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+
+	db "github.com/cosmos/cosmos-db"
+)
+
+// Cursors implements operations on the persisted cursors for network.
+type Cursors struct {
+	db              CursorTable
+	confirmInterval time.Duration
+	xProvider       xchain.Provider
+	streams         []stream
+	network         netconf.Network
+}
+
+func NewCursors(
+	db db.DB,
+	xProvider xchain.Provider,
+	confirmInterval time.Duration,
+	network netconf.Network,
+) (*Cursors, error) {
+	cursorsTable, err := NewCursorsTable(db)
+	if err != nil {
+		return nil, err
+	}
+
+	var streams []stream
```

Comment:
nit: cursors.streams is only used in one place, in `Monitor`, but `Monitor` has access to all the data to create it. So I'd suggest to remove it from `cursors` and just create them in `Monitor`